src/Diagnostics.RuntimeHost/Controllers/Site/SitesController.cs

```diff
@@ -68,16 +68,20 @@ public async Task<IActionResult> Post(string subscriptionId, string resourceGrou
         /// <param name="postBody">Request json body.</param>
         /// <returns>Task for listing detectors.</returns>
         [HttpPost(UriElements.Detectors)]
-        public async Task<IActionResult> ListDetectors(string subscriptionId, string resourceGroupName, string siteName, [FromBody] DiagnosticSiteData postBody)
+        public async Task<IActionResult> ListDetectors(string subscriptionId, string resourceGroupName, string siteName, [FromBody] DiagnosticSiteData postBody, [FromQuery(Name = "text")] string text = null)
         {
+            if (text == null)
+            {
+                text = Request.Query["text"].ToString();
```

Comment:
I didnt understand why we are checking for null and getting the value from `Request.Query["text"]`.  Aren't `FromQuery` and `Request.Query` the same?

Reply:
Removed